azurecontainerapps.ts

```diff
@@ -150,12 +157,8 @@ export class azurecontainerapps {
         // Get the YAML configuration file, if provided
         this.yamlConfigPath = this.toolHelper.getInput('yamlConfigPath', false) as string;
 
-        // Ensure that acrName or registryUrl is also provided if appSourcePath is provided
-        if (!this.util.isNullOrEmpty(this.appSourcePath) && this.util.isNullOrEmpty(this.acrName) && this.util.isNullOrEmpty(this.registryUrl)) {
-            let missingRegistryUrlMessage = `The 'acrName' or 'registryUrl' argument must be provided when the 'appSourcePath' argument is provided.`;
-            this.toolHelper.writeError(missingRegistryUrlMessage);
-            throw Error(missingRegistryUrlMessage);
-        }
+        // Get the name of the image to build if it was provided, or generate it from build variables
+        this.imageToBuild = this.toolHelper.getInput('imageToBuild', false);
```

Comment:
Could you please clarify why we needed to move this one out of `buildAndPushImageAsync`?

Reply:
We are leveraging the `imageToBuild` arg to determine if we want to use the CLI to build the image or this `buildAndPushImageAsync` method.  We need to move this out of this method so we can decide if we want to execute this method based on the condition below.

![image](https://github.com/Azure/container-apps-deploy-action/assets/108305436/652e55eb-f593-4cc4-be9f-e67936d5c403)